### PR TITLE
gnrc_tcp: improved option handling

### PR DESCRIPTION
**Pull request by brummer-simon**
- Removal of the criticized Option Field in the gnrc_tcp_hdr
- Header Cleanup

### CODE REVIEW

**Comment by OlegHahm** [comment]
Seems reasonable to me.

File: sys/include/net/gnrc/tcp/hdr.h [hunk 1]
```diff
@@ -29,24 +29,18 @@ extern "C" {
...
-} tcp_hdr_t;
+    network_uint16_t src_port;    /**< source port, in network byte order */
+    network_uint16_t dst_port;    /**< destination port, in network byte order */
+    network_uint32_t seq_num;     /**< sequence number, in network byte order */
+    network_uint32_t ack_num;     /**< Acknowledgement number, in network byte order */
+    network_uint16_t off_ctl;     /**< Data Offset and control Bits in network byte order */
+    network_uint16_t window;      /**< window, in network byte order */
+    network_uint16_t checksum;    /**< checksum, in network byte order */
+    network_uint16_t urgent_ptr;  /**< urgent pointer, in network byte order */
+} gnrc_tcp_hdr_t;
```
Comment (smlng) [inline sys/include/net/gnrc/tcp/hdr.h]: I'd rather leave it `tcp_hdr_t` and further move the header (similar to udp) to `sys/include/net/tcp.h`, as I already did in #6509. However, I agree to removing the options array.

File: sys/net/gnrc/transport_layer/tcp/gnrc_tcp_option.c [hunk 2]
```diff
@@ -33,58 +33,52 @@ uint16_t _option_build_offset_control(uint16_t nopts, uint16_t ctl)
...
+    uint8_t opt_left = (offset - OPTION_OFFSET_BASE) * sizeof(network_uint32_t);
 
-        /* If byte index is not aligned to word index. Fill word with bytes from next word. */
-        if (byte_idx) {
-            word >>= (byte_idx * 8);
-            word |= (byteorder_ntohl(hdr->options[word_idx + 1]) << ((sizeof(word) - byte_idx) * 8));
-        }
+    /* Parse Options Byte by Byte */
+    while (opt_left > 0) {
+        uint8_t kind = *opt_ptr;
```
Comment (smlng) [inline sys/net/gnrc/transport_layer/tcp/gnrc_tcp_option.c]: I would recommend to introduce option struct as follows:

```
struct tcp_opt {
    uint8_t kind,
    uint8_t len,
    uint8_t *value,
} tcp_opt_t;
```

File: sys/include/net/gnrc/tcp/hdr.h (hunk 1 above)
Comment (smlng) [inline sys/include/net/gnrc/tcp/hdr.h]: because the header is **not** GNRC specific, hence it should not reside exclusively inside GNRC.

**Comment by smlng** [comment]
please consider #6511

**Comment by brummer-simon** [comment]
Hi smlng, 

I have moved the tcp header to net/tcp.h and added your option parsing struct to net/tcp.h. The _parse_option Function uses the option struct now. 

Additionally I a bit of cleanup

**Comment by miri64** [comment]
The last commit seems unrelated? Why no new PR for that?

**Comment by brummer-simon** [comment]
Jeah I looked at the Files Section in this PR an will split this pr into two seperate pull requests tomorrow during Hack'n'Ack (I might need some help from the more experienced git users)

**Pull request closed without merging**